runtime/vm/jfr.cpp

```diff
@@ -739,8 +850,16 @@ jfrSamplingThreadProc(void *entryArg)
 		omrthread_monitor_enter(vm->jfrSamplerMutex);
 		vm->jfrSamplerState = J9JFR_SAMPLER_STATE_RUNNING;
 		omrthread_monitor_notify_all(vm->jfrSamplerMutex);
+		UDATA count = 0;
 		while (J9JFR_SAMPLER_STATE_STOP != vm->jfrSamplerState) {
 			J9SignalAsyncEvent(vm, NULL, vm->jfrAsyncKey);
+			if (count % 100 == 0) { // 1000ms
+				jfrCPULoad(currentThread);
+			}
+			if (count % 1000 == 0) { // 10s
+				J9SignalAsyncEvent(vm, NULL, vm->jfrThreadCPULoadAsyncKey);
```

Comment:
instead of using an async event, you should create a function that walks all threads and capture thread cpu load. The issue with asyncs is that if the thread is not active, it will never capture the async event. Given that this event runs evey 10s, we dont need to worry about the overhead of getting exclusive VMaccess

Reply:
One problem that I can think of is that `getrusage` can only be used to get info of the calling thread, so we may be not able to get user cpu time of other threads from the jfr thread?

Reply:
Yes, thats true. Okay ignore this comment for now.